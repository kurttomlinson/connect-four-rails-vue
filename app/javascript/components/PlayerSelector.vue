<template>
  <div class="player-selector">
    <div>Player {{ player }} type:</div>
    <input
      type="radio"
      :id="humanId"
      @change="radioButtonClick"
      :name="groupName"
      value="human"
      checked>
    <label :for="humanId">Human</label><br/>

    <input
      type="radio"
      :id="computerId"
      @change="radioButtonClick"
      :name="groupName"
      value="computer">
    <label :for="computerId">Computer</label><br/>
    <br/>
  </div>
</template>

<script>
import axios from 'axios';
// var $ = require('jquery');
// require('bootstrap');

export default {
  props: {
    player: {
      type: Number,
      required: true,
      validator: function (value) {
        return (value == 1) || (value == 2)
      }
    },
    playerType: {
      type: String,
      required: true,
      validator: function (value) {
        return (value == "human") || (value == "computer")
      }
    }
  },
  methods: {
    radioButtonClick (e, l) {
      this.$emit('playerSelected', this.player, e.target.value)
    }
  },
  computed: {
    groupName: function () {
      return "playerType" + this.player
    },
    humanId: function () {
      return "player" + this.player.toString() + "human"
    },
    computerId: function () {
      return "player" + this.player.toString() + "computer"
    }
  }
}
</script>

<style scoped>
</style>
