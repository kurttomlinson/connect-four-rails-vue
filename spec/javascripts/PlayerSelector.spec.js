import { shallow } from 'vue-test-utils'
import Vue from 'vue'
import PlayerSelector from 'components/PlayerSelector'

describe('PlayerSelector.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(PlayerSelector, {
      propsData: {
        player: 1,
        playerType: 'human'
      }
    })
  })

  it('emits a "setPlayerType" event when "computer" is picked', () => {
    const stub = jest.fn()
    wrapper.vm.$on('setPlayerType', stub)
    wrapper.find('#player1computer').trigger('click')
    expect(stub).toBeCalledWith(1, 'computer')
  })

  it('emits a "setPlayerType" event when "human" is picked', () => {
    const stub = jest.fn()
    wrapper.vm.$on('setPlayerType', stub)
    wrapper.find('#player1computer').trigger('click')
    wrapper.find('#player1human').trigger('click')
    expect(stub).toBeCalledWith(1, 'human')
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
