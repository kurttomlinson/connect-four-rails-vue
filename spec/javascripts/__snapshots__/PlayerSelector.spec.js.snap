// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerSelector.vue has the expected html structure 1`] = `
<div
  class="player-selector"
>
  <div>
    Player 1 type:
  </div>
   
  <input
    checked="checked"
    id="player1human"
    name="playerType1"
    type="radio"
    value="human"
  />
   
  <label
    for="player1human"
  >
    Human
  </label>
  <br />
   
  <input
    id="player1computer"
    name="playerType1"
    type="radio"
    value="computer"
  />
   
  <label
    for="player1computer"
  >
    Computer
  </label>
  <br />
   
  <br />
</div>
`;
